config/initializers/content_security_policy.rb

```diff
@@ -19,6 +19,20 @@ def host_to_url(str)
 media_host ||= host_to_url(ENV['S3_HOSTNAME']) if ENV['S3_ENABLED'] == 'true'
 media_host ||= assets_host
 
+def sso_host
+  return if ENV['OMNIAUTH_ONLY'] != 'true' || Devise.omniauth_providers.length != 1
+
+  provider = Devise.omniauth_configs[Devise.omniauth_providers[0]]
+  @sso_host ||= begin
+    # using CAS
+    provider.cas_url if ENV['CAS_ENABLED'] == 'true'
+    # using SAML
+    provider.options[:idp_sso_target_url] if ENV['SAML_ISSUER'] == 'true'
```

Comment:
You probably meant `SAML_ENABLED` rather than `SAML_ISSUER`, right?
```suggestion
    provider.options[:idp_sso_target_url] if ENV['SAML_ENABLED'] == 'true'
```